### PR TITLE
Add v0.23.0 to releases.md

### PR DESCRIPTION
**Pull request by khrm**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind misc

File: releases.md [hunk 1]
```diff
@@ -40,20 +40,20 @@ Further documentation available:
 
 ## Releases
 
+### v0.23
+
+- **Latest Release**: [v0.23.0][v0-23-0] (2023-03-02) ([docs][v0-23-0-docs], [examples][v0-23-0-examples])
```
Comment (khrm) [inline releases.md:45]: In the preview, the link aren't working. What could be the reason?

Comment (khrm) [reply releases.md:45]: Yes, that's the issue. Thanks.

**Review by savitaashture — APPROVED** [review]
/lgtm

**Pull request merged**